tests/integration_tests/with_db/fixtures/setup_database_fixture.py

```diff
@@ -1,43 +1,98 @@
 
 
-def _deploy_scripts() -> None:
+def _create_schema(db_conn) -> str:
+    schema = SCHEMA_NAME
+    db_conn.execute(f"DROP SCHEMA IF EXISTS {schema} CASCADE;")
+    db_conn.execute(f"CREATE SCHEMA IF NOT EXISTS {schema};")
+    return schema
+
+
+def _deploy_scripts(db_conn) -> None:
     ScriptsDeployer.run(
-        dsn=db_params.address(),
-        user=db_params.user,
-        password=db_params.password,
-        schema=schema_name,
-        language_alias=language_alias,
+        db_conn,
+        schema=SCHEMA_NAME,
+        language_alias=LANGUAGE_ALIAS,
         develop=True)
 
+# Can be removed after 
+# https://github.com/exasol/advanced-analytics-framework/issues/176
+def _bucket_address(
+        bucketfs_params: dict[str, Any],
+        path_in_bucket: str = "my-folder",
+) -> str:
+    url = bucketfs_params["url"]
+    bucket_name = bucketfs_params["bucket_name"]
+    service_name = bucketfs_params["service_name"]
+    return ( f"{url}/{bucket_name}/"
+             f"{path_in_bucket};{service_name}" )
+
+
+# Can be removed after 
+# https://github.com/exasol/advanced-analytics-framework/issues/176
+def _create_bucketfs_connection(
+        use_onprem,
+        db_conn,
+        bucketfs_params: dict[str, Any],
+        name: str = BUCKETFS_CONNECTION_NAME,
+        path_in_bucket: str = "my-folder",
+) -> str:
+    if use_onprem:
+        uri = _bucket_address(bucketfs_params, path_in_bucket)
+        user = bucketfs_params["username"]
+        pwd = bucketfs_params["password"]
+        db_conn.execute(
+            f"CREATE OR REPLACE  CONNECTION {name} TO '{uri}' " \
+            f"USER '{user}' IDENTIFIED BY '{pwd}'"
+        )
+    return name
+
 
-def _create_bucketfs_connection(db_conn) -> None:
-    query = "CREATE OR REPLACE  CONNECTION {name} TO '{uri}' " \
-            "USER '{user}' IDENTIFIED BY '{pwd}'".format(
-        name=bucketfs_connection_name,
-        uri=bucketfs_params.address(bucketfs_params.real_port),
-        user=bucketfs_params.user,
-        pwd=bucketfs_params.password)
-    db_conn.execute(query)
+# Can be removed after 
+# https://github.com/exasol/advanced-analytics-framework/issues/176
+@pytest.fixture(scope='session')
+def my_bucketfs_connection_factory(
+        use_onprem,
+        pyexasol_connection,
+        backend_aware_bucketfs_params,
+) -> Callable[[str, str|None], None]:
+    def create(name, path_in_bucket):
+        _create_bucketfs_connection(
+            use_onprem,
+            pyexasol_connection,
+            backend_aware_bucketfs_params,
+            name,
+            path_in_bucket,
+        )
+    return create
 
 
 @pytest.fixture(scope="module")
-def setup_database(pyexasol_connection) -> Tuple[str, str]:
-    _create_schema(pyexasol_connection)
-    _deploy_scripts()
-    _create_bucketfs_connection(pyexasol_connection)
-    return bucketfs_connection_name, schema_name
+def database_with_slc(
+        use_onprem,
```

Comment:
You don't use the `use_onprem` fixture here, do you?

Reply:
Thanks - removed in next push.